purchase_work_acceptance/models/account_invoice.py

```diff
@@ -0,0 +1,169 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+from odoo.tools.float_utils import float_compare
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    wa_id = fields.Many2one(
+        comodel_name='work.acceptance',
+        string='WA Reference',
+        domain=[('state', '=', 'accept')],
+        store=True,
+    )
+
+    @api.multi
+    def action_invoice_open(self):
+        self.ensure_one()
+        if self.wa_id:
+            wa_lines = self.env['work.acceptance.line']
+            for line in self.invoice_line_ids:
+                id = line.purchase_line_id.id
+                wa_line = wa_lines.search([('purchase_line_id', '=', id)])
+                if line.quantity > wa_line.product_qty:
```

Comment:
Let's check for "=", not ">".
And let's check both quantity and price_unit, shall we?